Python_based_scripts/Plotting_ABS_Census_Data_LGA_2021.ipynb

```diff
@@ -0,0 +1,5188 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Extracting temperature at specific reefs\n",
+    "**Author**: Denisse Fierro Arcos  \n",
+    "\n",
+    "This notebook will show how to access the RIMReP `geoparquet` collection for the Regional Data by Local Government Area (LGA) from the Australian Bureau of Statistics (ABS), which contains data for the period between 2011 and 2022. This dataset includes data for the Census of Population and Housing for 2011, 2016 and 2021.\n",
+    "\n",
+    "Although this dataset has information available for all Australian states and territories, we will focus on accessing data for the state of Queensland only. We will create some summary table, as well as figures and maps."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Accessing S3 bucket\n",
+    "from pyarrow import parquet as pq\n",
+    "import s3fs\n",
+    "# Manipulating dataframes\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import copy\n",
+    "# Checking patterns in strings using regular expressions (regex)\n",
+    "import re\n",
+    "# Interacting with operating system\n",
+    "import os\n",
+    "# Load and manipulate geospatial data\n",
+    "import dask_geopandas\n",
+    "import geopandas as gpd\n",
+    "# Plotting\n",
+    "import matplotlib.pyplot as plt\n",
+    "#Color maps\n",
+    "from matplotlib import colormaps as cmaps\n",
+    "# Mapping\n",
+    "import cartopy.crs as ccrs\n",
+    "import cartopy.feature as cf"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Connecting to ABS regional dataset in RIMReP collection"
```

Comment:
this section doesn't work for me:

ArrowInvalid: GetFileInfo() yielded path 'rimrep-data-public/abs-regional-lga-2021/data.parquet/part.0.parquet', which is outside base dir 's3://rimrep-data-public/abs-regional-lga-2021/data.parquet'

Reply:
I have removed this. It seems this option is a bit fiddly.

Reply:
That error was because when using the s3fs filesystem you need to remove the "s3://" part from the URL of the dataset. That is:

`md = pq.ParquetDataset('rimrep-data-public/abs-regional-lga-2021/data.parquet', filesystem = fs)` should work for everyone, while

`md = pq.ParquetDataset('s3://rimrep-data-public/abs-regional-lga-2021/data.parquet')` only works if you don't have any environment variables related to AWS set up.

`md = pq.ParquetDataset('s3://rimrep-data-public/abs-regional-lga-2021/data.parquet', filesystem = fs)` won't work for anyone, though. @lidefi87